lib/rouge/theme.rb

```diff
@@ -135,7 +135,9 @@ def registry
 
   module HasModes
     def mode(arg=:absent)
-      return @mode if arg == :absent
+      if arg == :absent
+        return (defined?(@mode) ? @mode : nil)
+      end
```

Comment:
this seems excessive...

Reply:
I agree in principle but was the simplest way I could think of to fix the warning.

Reply:
Again, I'm actually completely okay with rouge depending on uninitialized instance variables. This code is basically a fancy version of `||=`, which ruby explicitly supports.

Reply:
I don't run Rouge with warnings turned on so I don't really care but for a library like Rouge that's typically part of a larger project (eg. Jekyll, GitLab, etc), being able to run without warnings seems like a nice feature.

Reply:
Oh, and in case it helps, here's the error for this one:

```
lib/rouge/theme.rb:138: warning: instance variable @mode not initialized
```